backend/functions/utils.py

```diff
@@ -0,0 +1,187 @@
+import requests
+from rest_framework import status
+from dataset import models as dataset_models
+from google.cloud import translate_v2 as translate
+from users.utils import (
+    LANG_NAME_TO_CODE_GOOGLE,
+    LANG_NAME_TO_CODE_ULCA,
+    LANG_TRANS_MODEL_CODES,
+)
+
+
+### Utility Functions
+def check_translation_function_inputs(
+    input_dataset_instance_id, output_dataset_instance_id
+):
+
+    """Function to check the input parameters for the translation function. This performs checks on input dataset instance and output dataset instance.
+
+    Returns:
+        input_dataset_instance_id: ID of the input dataset which has to be a SentenceText DatasetInstance.
+        output_dataset_instance_id: ID of the output dataset which has to be a TranslationPair DatasetInstance.
+    """
+
+    # Check if the input dataset instance is a SentenceText dataset
+    try:
+        input_dataset_instance = dataset_models.DatasetInstance.objects.get(
+            instance_id=input_dataset_instance_id
+        )
+
+        # Check if it is a sentence Text
+        if input_dataset_instance.dataset_type != "SentenceText":
+            return {
+                "message": "Input dataset instance is not a SentenceText dataset.",
+                "status": status.HTTP_400_BAD_REQUEST,
+            }
+
+    except dataset_models.DatasetInstance.DoesNotExist:
+        return {
+            "message": "Dataset instance does not exist!",
+            "status": status.HTTP_404_NOT_FOUND,
+        }
+
+    # Check if the output dataset instance exists and is a TranslationPair type
+    try:
+        output_dataset_instance = dataset_models.DatasetInstance.objects.get(
+            instance_id=output_dataset_instance_id
+        )
+        if output_dataset_instance.dataset_type != "TranslationPair":
+            return {
+                "message": "Output dataset instance is not of type TranslationPair",
+                "status": status.HTTP_400_BAD_REQUEST,
+            }
+
+    except dataset_models.DatasetInstance.DoesNotExist:
+        return {
+            "message": "Output dataset instance does not exist! Create a TranslationPair DatasetInstance",
+            "status": status.HTTP_404_NOT_FOUND,
+        }
+
+    return {"message": "Success", "status": status.HTTP_200_OK}
+
+
+def get_batch_translations_using_indictrans_nmt_api(
+    sentence_list, source_language, target_language
+):
+
+    """Function to get the translation for the input sentences using the IndicTrans NMT API.
+
+    Args:
+        sentence_list (str): List of sentences to be translated.
+        source_language (str): Original language of the sentence.
+        target_language (str): Final language of the sentence.
+
+    Returns:
+        list: List of dictionaries containing the translated sentences.
+    """
+
+    # Get the translation model ID
+    model_id = LANG_TRANS_MODEL_CODES.get(f"{source_language}-{target_language}", 144)

```

Comment:
Instead of hardcoding `144`, can you please create a global variable, something like `DEFAULT_ULCA_INDIC_MODEL_ID` and use that in both the functions: `get_batch_translations_using_indictrans_nmt_api()` and `get_translation_using_cdac_model()`

Reply:
Sure, I will make that change.